app/models/rpc/DSAMessage.scala

```diff
@@ -53,4 +55,15 @@ case class ResponseMessage(msg: Int, ack: Option[Int] = None, responses: List[DS
     s"ResponseMessage($msg,$ack,$responses)"
   else
     s"ResponseMessage($msg,$ack,List(${responses.head},...${responses.size - 1} more))"
+}
+
+case class SubscriptionNotificationMessage(msg: Int, ack: Option[Int] = None, responses: List[DSAResponse] = Nil, sid: Int, qos: QoS.Level = QoS.Default)
```

Comment:
I don't think `SubscriptionNotificationMessage` should extends `DSAMessage`. `DSAMessage` defines DSA communication protocol, while the former is purely implementation-specific.

Reply:
Moved to Message